pkg/build/api/validation/validation.go

```diff
@@ -51,8 +51,8 @@ func ValidateBuildConfig(config *buildapi.BuildConfig) fielderrors.ValidationErr
 			}
 		}
 	}
-	allErrs = append(allErrs, validateBuildParameters(&config.Parameters).Prefix("parameters")...)
-	allErrs = append(allErrs, validateBuildConfigOutput(&config.Parameters.Output).Prefix("parameters.output")...)
+	allErrs = append(allErrs, validateBuildParameters(&config.Parameters).Prefix("spec")...)
+	allErrs = append(allErrs, validateBuildConfigOutput(&config.Parameters.Output).Prefix("spec.output")...)
```

Comment:
this is what surfaces to API users to tell them what's wrong with what they submitted. it is supposed to match the serialized field name